WordPress/Classes/Services/BlogJetpackSettingsService.swift

```diff
@@ -0,0 +1,210 @@
+            var remoteJetpackMonitorSettings: RemoteBlogJetpackMonitorSettings? = nil
+
+            // Create a dispatch group to wait for both calls.
+            let syncGroup = DispatchGroup()
+
+            syncGroup.enter()
+            remote.getJetpackSettingsForSite(blogDotComId,
+                                             success: { (remoteSettings) in
+                                                remoteJetpackSettings = remoteSettings
+                                                successfulFetches += 1
+                                                syncGroup.leave()
+                                             }, failure: { (error) in
+                                                fetchError = error
+                                                syncGroup.leave()
+                                             })
+
+            syncGroup.enter()
+            remote.getJetpackMonitorSettingsForSite(blogDotComId,
+                                                    success: { (remoteMonitorSettings) in
+                                                        remoteJetpackMonitorSettings = remoteMonitorSettings
+                                                        successfulFetches += 1
+                                                        syncGroup.leave()
+                                                    }, failure: { (error) in
+                                                        fetchError = error
+                                                        syncGroup.leave()
+                                                    })
+
+            syncGroup.notify(queue: DispatchQueue.main, execute: {
+                guard let remoteJetpackSettings = remoteJetpackSettings,
+                    let remoteJetpackMonitorSettings = remoteJetpackMonitorSettings else {
+                        failure(nil)
+                        return
+                }
+                if successfulFetches == 2 {
+                    self.context.perform {
+                        self.updateJetpackSettings(blogSettings, remoteSettings: remoteJetpackSettings)
+                        self.updateJetpackMonitorSettings(blogSettings, remoteSettings: remoteJetpackMonitorSettings)
+                    }
+                    do {
+                        try self.context.save()
+                        success()
+                    } catch let error as NSError {
+                        failure(error)
+                    }
+                } else {
+                    failure(fetchError)
+                }
+            })
+        }
+    }
+
+    // Jetpack settings have to be updated one by one because the API does not allow us to do it in any other way.
+
+    func updateJetpackMonitorEnabledForBlog(_ blog: Blog, value: Bool, success: @escaping () -> Void, failure: @escaping (Error?) -> Void) {
+        updateJetpackSettingForBlog(blog,
+                                    key: BlogJetpackSettingsServiceRemote.Keys.monitorEnabledKey,
```

Comment:
This might look nicer if `BlogJetpackSettingsServiceRemote.Keys` were an enum, but I'm not sure if the remote code would then be messier. Did you try this already?

Reply:
I had tried it and it was giving me problems on the local side of things when I was still using Obj-C (long story). I've changed it back to an Enum now and it all seems to be working. Thanks for the suggestion.

Reply:
👍 this is better. What I meant originally was to use enum values for the keys, not just an enum as a namespace (instead of a struct), so you could tweak the method and drop the `BlogJetpackSettingsServiceRemote.Keys` prefix on every instance. However this would also mean adding a bunch of `.rawValue` in the remote code, so I think it's better as it is now.